optuna/storages/_base.py

```diff
@@ -189,6 +190,11 @@ def set_study_directions(self, study_id: int, directions: Sequence[StudyDirectio
         """
         raise NotImplementedError
 
+    def get_study_class(self) -> Type["optuna.study.Study"]:
+        from optuna.study import Study
+
+        return Study
+
```

Comment:
I wanted to briefly highlight this change as it impacts the `BaseStorage` class. This `get_study_class` method was added so that we can have `create_study` return a `DaskStudy` when a `DaskStorage` instance is passed to `create_study`:

https://github.com/optuna/optuna/blob/30d10684a9c07ec476fbb4cf59710463f5161608/optuna/study/study.py#L1155-L1156

Note that in the common case `get_study_class` will return the normal `Study` object. It's up to `BaseStorage` subclasses (like `DaskStorage`) to specific overload `get_study_class` if they'd like some other study class to be returned. 

This seemed like a minimal enough extension to allow `create_study` to return other study subclasses, but I'm certainly open to suggestions from others

Reply:
Having seamless first class support might be ideal but I'm a bit concerned that this is somewhat intrusive. Ideally speaking, we shouldn't further couple studies and storages. What do you think about instead having `DaskStudy` take a `study: Study` in its `DaskStudy.__init__` and having the user "convert" a `Study` to `DaskStudy`? It'll somewhat resemble the `TorchDistributedTrial` https://github.com/optuna/optuna/pull/2303/files#diff-6444895310850f4fc473d85e441b9db0c311f7c3b77f5e63506afc530f356eb6R91. Can we in that way provide the Dask functionality without changing the core part the framework?